packages/react-native-audio-api/android/src/main/java/com/swmansion/audioapi/system/notification/PlaybackNotification.kt

```diff
@@ -0,0 +1,578 @@
+      }
+      return buildNotification()
+    }
+
+    // Update metadata
+    if (options.hasKey("title")) {
+      title = options.getString("title")
+    }
+
+    if (options.hasKey("artist")) {
+      artist = options.getString("artist")
+    }
+
+    if (options.hasKey("album")) {
+      album = options.getString("album")
+    }
+
+    if (options.hasKey("duration")) {
+      duration = (options.getDouble("duration") * 1000).toLong()
+    }
+
+    if (options.hasKey("elapsedTime")) {
+      elapsedTime = (options.getDouble("elapsedTime") * 1000).toLong()
+    } else {
+      // Use the current position from the media session controller (live calculated position)
+      val controllerPosition = mediaSession?.controller?.playbackState?.position
+      if (controllerPosition != null && controllerPosition > 0) {
+        elapsedTime = controllerPosition
+      }
+    }
+
+    if (options.hasKey("speed")) {
+      speed = options.getDouble("speed").toFloat()
+    } else {
+      // Use the current speed from the media session controller
+      val controllerSpeed = mediaSession?.controller?.playbackState?.playbackSpeed
+      if (controllerSpeed != null && controllerSpeed > 0) {
+        speed = controllerSpeed
+      }
+    }
+
+    // Ensure speed is at least 1.0 when playing
+    if (isPlaying && speed == 0f) {
+      speed = 1.0f
+    }
+
+    // Update playback state
+    if (options.hasKey("state")) {
+      when (options.getString("state")) {
+        "playing" -> {
+          playbackPlayingState = PlaybackStateCompat.STATE_PLAYING
+        }
+        "paused" -> {
+          playbackPlayingState = PlaybackStateCompat.STATE_PAUSED
+        }
+      }
+    }
+
+    // Build MediaMetadata
+    val metadataBuilder =
+      MediaMetadataCompat
+        .Builder()
+        .putString(MediaMetadataCompat.METADATA_KEY_TITLE, title)
+        .putString(MediaMetadataCompat.METADATA_KEY_ARTIST, artist)
+        .putString(MediaMetadataCompat.METADATA_KEY_ALBUM, album)
+        .putLong(MediaMetadataCompat.METADATA_KEY_DURATION, duration)
+
+    // Update notification builder
+    notificationBuilder?.setContentTitle(title)
+    notificationBuilder?.setContentText(artist)
+
+    // Handle artwork
+    if (options.hasKey("artwork")) {
+      artworkThread?.interrupt()
+
+      val artworkUrl: String?
+      val isLocal: Boolean
+
+      if (options.getType("artwork") == ReadableType.Map) {
+        artworkUrl = options.getMap("artwork")?.getString("uri")
+        isLocal = true
+      } else {
+        artworkUrl = options.getString("artwork")
+        isLocal = false
+      }
+
+      if (artworkUrl != null) {
+        artworkThread =
+          Thread {
+            try {
+              val bitmap = loadArtwork(artworkUrl, isLocal)
+              if (bitmap != null) {
+                artwork = bitmap
+                metadataBuilder.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, bitmap)
+                notificationBuilder?.setLargeIcon(bitmap)
```

Comment:
currently there is no api for setting small icon as requested in #820,

```suggestion
notificationBuilder
     ?.setLargeIcon(bitmap)    
     ?.setSmallIcon(IconCompat.createWithBitmap(bitmap))
```
this would set up small icon to the same value as large icon